token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyService.java

```diff
@@ -50,10 +50,10 @@ public String retrieveTokenKeys(URI tokenKeysEndpointUri, @Nullable String zoneI
 		request.addHeader(HttpHeaders.USER_AGENT, HttpClientUtil.getUserAgent());
 
 		LOGGER.debug("Executing token key retrieval GET request to {} with headers: {} ", tokenKeysEndpointUri,
-				request.getAllHeaders());
-		try (CloseableHttpResponse response = httpClient.execute(request)) {
+				request.headerIterator());
+		try (ClassicHttpResponse response = httpClient.execute(request)) {
```

Comment:
Since Apache HTTP Client 5, one should use non-deprecated execute method by supplying a Response Handler. As discussed, we will overhaul these calls in the next commits.